projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-site-settings-endpoint.php

```diff
@@ -499,6 +499,29 @@ public function get_settings_response() {
 
 	}
 
+	/**
+	 * Get the default value for the wpcom_gifting_subscription option.
+	 * The default value is the inverse of the plan's auto_renew setting.
+	 *
+	 * @return bool
+	 */
+	protected function get_wpcom_gifting_subscription_default() {
+		if ( function_exists( 'wpcom_get_site_purchases' ) && function_exists( 'wpcom_purchase_has_feature' ) ) {
+			$purchases = wpcom_get_site_purchases();
+
+			foreach ( $purchases as $purchase ) {
+				if ( wpcom_purchase_has_feature( $purchase, \WPCOM_Features::SUBSCRIPTION_GIFTING ) ) {
+					if ( isset( $purchase->auto_renew ) ) {
+						return ! $purchase->auto_renew;
+					} elseif ( isset( $purchase->user_allows_auto_renew ) ) {
+						return ! $purchase->user_allows_auto_renew;
+					}
```

Comment:
Checking for these different versions of `auto_renew` is an ugly consequence of this issue in D91785-code #inline-333930. Hopefully, we'll get that resolved at some point. Meanwhile, I tried to write this to work safely in either case.

Reply:
I think we can fix this easier when we move it all into Jetpack.